### ISSUE TITLE
Board View should be able to group by List (tags)

### ISSUE THREAD

**Issue by nunocasteleira** [marcusolsson/obsidian-projects#117]
### What would you like to be added?

When adding a Board view, we should be able to group notes by tags.

### Why is this needed?

AS tags can be easily used to mark related notes, they should be used as a grouping option.

**Comment by marcusolsson** [comment]
Could you elaborate on how you'd expect the feature to work?

If I understand correctly, you'd like to group by a list field instead of a string field?

For example, a note can have multiple tags. Would you expect the note to be present in all the columns?

**Comment by nunocasteleira** [comment]
Correct, didn't think with that constraint in mind. Would it make sense to present the note in both columns?

**Comment by marcusolsson** [comment]
Whether it makes sense would depend on the use case. Might be better as a separate view.

**Comment by nunocasteleira** [comment]
I'm available to help with this feature and I will have a look on other features also 😊

**Comment by LeslieP-TX** [comment]
Maybe not grouping by ANY tag, but if we use #story/idea #story/draft #story/final #story/published, being able to group by #story and have a separate lane for each subtag would be great.

**Comment by pwia** [comment]
Another benefit of including non-text fields is the ability to leverage the links in Obsidian.  For instance, in a text field it seems only one [[file]] can be linked, whereas in a list type, multiple obsidian links can be used.  This could be leveraged for sorting features.  In my use cases, this is superior to tags.  However, the general idea would be similar I think.

An example would be for a sales lead database (similar to a CRM database functionality).  Perhaps I have a number of meetings with different customers and make a note for each meeting. I would have a "customer name" and a "product" property field with type = list.  I track the "customer name" interested in products within my portfolio.  Most of these customers would be interested in more than one "product." So, I use a list-type and include each product name as its obsidian [[link]].  Ideally, I'd then like to be able to sort the board status by the "product" property field by the product name in a board view.  This would then give me columns of each product and the customers interested.

I'm an in-experienced programmer and thus can offer little support in this manner, but I'm happy to help support evaluation if some fresh eyes would be helpful. 

Thanks for your work on this plugin, it is very useful.

**Comment by jan-chvojka** [comment]
I've just start to use Obsidian and Projects so maybe this will be wrong assumption but when I seen status as a text field I immediately changed to the list because then I can choose from all statuses when editing note. 
I was surprised that board columns are gone by changing filed type. I think that note should appear in column by first item of the list but I think that user just can pay attention to have one value in the field. Both ways would work for me.

**Comment by lightningRalf** [comment]
> Could you elaborate on how you'd expect the feature to work?
> 
> If I understand correctly, you'd like to group by a list field instead of a string field?
> 
> For example, a note can have multiple tags. Would you expect the note to be present in all the columns?

I have all my properties lined up as lists:

Examples:
#### [[Natural planning model|NPM]]-StatusList - Anstehende Aktion - gewählt ohne Tag aufgrund von Projects Arbeitsweise:
`toProcess`
`toBrainstorm`
`toOrganize`
`toPlan`
`toExecute`
`done`
`cancelled`
#### Projektstatus
`01-new`
`02-icebox`
`03-backlog`
`04-inProgress`
`05-inReview`
`06-done`
##### Dringlichkeit: == zeitbezogen
`21-someDay`
`13-thisYear`
`8-thisQuarter`
`5-thisMonth`
`3-thisWeek`
`2-Today`
`1-now`
##### Wichtigkeit: == einflussreich 
`13-lowest`
`08-low`
`05-normal`
`03-medium`
`02-high`
`01-highest`

So having lists available, would solve a lot of pain points for me. Would be huge if this could be included. 
I think #849 mentions the same thing.

**Comment by lightningRalf** [comment]
#853 also mentions this apparently.

And the change could be marginly small?
https://copilot-workspace.githubnext.com/marcusolsson/obsidian-projects?shareId=884ad9ac-1ec0-46fe-bc15-76414caab273
Or am I missing something? @Acylation

**Comment by Vadych** [comment]
I am also interested in solving this problem.  Of course, I would prefer the note to be in only one column. You can select the first value in the multiple list. Ideally, when dragging, replace the previous first value with the new one.

**Comment by SirArCade** [comment]
I am also interested in this problem. And I would prefer that the note is displayed in all columns to which it relates, it seems to me the most logical.

Hypothetical example: I have a library in which each work is indicated by the author. One work can have several authors due to work in a duet or in a team (if it is a collection or anthology)
And so, I want to sort the library by authors to see how many and which works of each author I have managed to get acquainted with. In this case, of course, I want the work created in a duet to be displayed for both authors.

In addition to the library, this can also be useful in projects, when one note equally relates to several areas of the project
